test/unit/select.js

```diff
@@ -0,0 +1,66 @@
+// TODO(naguib): Fail tests if gas utilization changes
+// TODO(naguib): Consider Adding more test scenarios
```

Comment:
Any more scenarios needed? Can we add them or remove this comment if not?

Reply:
Removed.